UXIT-Contractors/UXIT-Planner-App
# Day/feat/cover absence

### Title: 
Feature: Cover Absence Functionality

### Description:
This pull request introduces the "Cover Absence" functionality into the UXIT-Planner-App. This feature allows team members to cover the shifts of their colleagues who are absent, providing better flexibility and cooperation within the team. 

Key changes include:

- The introduction of new models: EvenWeek, OddWeek, and ReservationWeek to schema.prisma.
- Improvements to backup generation, with the system now generating two backups per day.
- The creation of an "Absence" page, which provides a user interface for managing absences.
- The implementation of the "cover a shift" feature, which allows users to volunteer to cover the shifts of absent colleagues.

### Testing:

To test these changes, follow these steps:

1. Pull the changes from the `day/feat/cover-absence` branch.
2. Make sure to build the project and update the database schema with the new models, and log out and in again after.
3. Visit the new `open-shift/birthday` page and familiarize yourself with its interface. (this page works like `open-shift/[id]` but i included the birthday to be able to test)
4. Test the "Neem shift" functionality by clicking the button, the open-shift should be gone from the database and appeared in your schedule on the right day.

### Request:

Please review the changes, test the new functionalities as described above, and if everything is satisfactory, merge this branch into the master branch.
